### PR TITLE
Environment db__url --> db.url; Add to environment.cfg file

### PR DESCRIPTION
**Pull request by terrywbrady**
If this change looks good, the config-definition.xml file should be added to the DSpace 6 and DSpace 7 dockerfiles.

Alternative variable passing approach for https://github.com/DSpace-Labs/DSpace-Docker-Images/pull/92

### CODE REVIEW

File: add-ons/dspace-docker-tools/ingestAIP.sh [hunk 1]
```diff
@@ -2,6 +2,8 @@
 # Once this file has been saved to a docker volume, the ingest step will not be re-run
 CHECKFILE=/dspace/assetstore/ingest.hasrun.flag
 
+env | egrep "__.*=" | egrep -v "=.*__" | sed -e s/__/./g > /dspace/config/environment.cfg
```
Comment (mwoodiupui) [inline add-ons/dspace-docker-tools/ingestAIP.sh]: This pipe excludes any variable whose *value* contains a double underscore.  That's highly unlikely to cause problems, but I did work out a different one that doesn't have that issue:
``env | awk -F = -e '/__.*=/ { name = $1; divide = index($0, "="); value = substr($0, divide+1); gsub(/__/, ".", name); print name "=" value }' > /dspace/config/environment.cfg``

File: docker-compose-files/dspace-compose/config-definition.xml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+             e.g. ${sys:dspace.dir} or ${env:DSPACE_HOME}
+             For more info see:
+             https://commons.apache.org/proper/commons-configuration/userguide/howto_basicfeatures.html#Variable_Interpolation -->
+        <!-- Load Java System properties (does not automatically reload) -->
+        <system/>
+        <!-- Load Environment variables (does not automatically reload) -->
+        <env/>
+
+        <!-- Allows docker insertion of properties containing periods from environment variables -->
+        <properties fileName="environment.cfg" throwExceptionOnMissing="false" config-optional="true"
```
Comment (tdonohue) [inline docker-compose-files/dspace-compose/config-definition.xml]: @terrywbrady : I have to admit, I'm a little hesitant to add an entirely new configuration file (`environment.cfg`) *just for* the Docker `ingestAIP.sh` script (which seems to be the only place this is used).  

I worry it's a bit complex to have two overriding config files (`local.cfg` and now `environment.cfg`) as DSpace users would need to clearly understand precedence of each. They'd also need to realize that an "environment.cfg" file in Docker will *override* any changes they try to make in the "local.cfg" -- which could get very frustrating, if they are used to changing "local.cfg" and find Docker isn't respecting those changes any longer.

Also, it seems like `ingestAIP.sh` script only gets run *once*, which would imply that changes to environment variables are not being reloaded into your `environment.cfg`.  So, after initial startup of Docker, you would need to modify the `environment.cfg` file directly, and ignore anything set in environment variables.

I'm sorry to say this, but honestly, if we only really need to customize ~15 settings (`dspace.name`, `db.*` settings, and `rdf.*` settings) during Docker startup, then I still feel we may be better off using `JAVA_OPTS`.  I'm worried about this approach for the reasons mentioned above.

**Comment by terrywbrady** [comment]
Closing based on the conversation above.

Tim Donohue [7:59 AM]
@terrywbrady & @mwood:  Sorry, but that approach worries me a bit.  I've added a comment inline here to describe why: https://github.com/DSpace-Labs/DSpace-Docker-Images/pull/93/files#r265170629   (Essentially, I'm worried we are creating non-standard DSpace behavior in our Docker images, which doesn't seem like a good direction to me) (edited) 


Terry Brady [8:52 AM]
@tdonohue, I appreciate the review and the caution on this.  I recommend that we move forward with the system properties approach for now.  If we hit complications with that approach, I think we have a good starting point for a solution.  If we do need greater flexibility, one alternative could be for the process to overwrite local.cfg within the container rather than writing to a new environment.cfg file.

Tim Donohue [8:56 AM]
@terrywbrady: That sounds good to me.  Yes, I'd agree that overwriting the existing `local.cfg` seems better -- as it's expected behavior that local.cfg includes all your overridden properties anyhow.  So, that sounds like a better backup option to me.

**Comment by terrywbrady** [comment]
@mwoodiupui , @tdonohue , @pnbecker ,

I had one additional thought...

One limitation of the system properties approach is that a single string is passed in, so the last file to set the string will determine the properties that are passed.

With an approach such as the one outlined in this PR, a map of env variables can be accumulated from each individual override file.  Each override file can neatly package the override variables on which it depends.

**Comment by terrywbrady** [comment]
## one.yml

```
version: "3.7"

services:
  dspace:
    image: "${DOCKER_OWNER:-dspace}/dspace:${DSPACE_VER:-dspace-6_x-jdk8-test}"
    environment:
      - JAVA_OPTS=one
```

## two.yml

```
version: "3.7"

services:
  dspace:
    image: "${DOCKER_OWNER:-dspace}/dspace:${DSPACE_VER:-dspace-6_x-jdk8-test}"
    environment:
      - JAVA_OPTS=two
```

## three.yml

```
version: "3.7"

services:
  dspace:
    image: "${DOCKER_OWNER:-dspace}/dspace:${DSPACE_VER:-dspace-6_x-jdk8-test}"
    environment:
      - JAVA_OPTS=${JAVA_OPTS:-def} three
```

## `docker-compose -f one.yml -f two.yml -f three.yml config`

```
services:
  dspace:
    environment:
      JAVA_OPTS: def three
    image: dspace/dspace:dspace-6_x-jdk8-test
version: '3.7'

```

**Pull request closed without merging**